src/main/kotlin/no/nav/tilleggsstonader/sak/interntVedtak/InterntVedtakTask.kt

```diff
@@ -0,0 +1,70 @@
+package no.nav.tilleggsstonader.sak.interntVedtak
+
+import no.nav.familie.prosessering.AsyncTaskStep
+import no.nav.familie.prosessering.TaskStepBeskrivelse
+import no.nav.familie.prosessering.domene.Task
+import no.nav.tilleggsstonader.kontrakter.dokarkiv.ArkiverDokumentRequest
+import no.nav.tilleggsstonader.kontrakter.dokarkiv.Dokument
+import no.nav.tilleggsstonader.kontrakter.dokarkiv.Filtype
+import no.nav.tilleggsstonader.kontrakter.dokarkiv.dokumentTypeInterntVedtak
+import no.nav.tilleggsstonader.sak.arbeidsfordeling.ArbeidsfordelingService
+import no.nav.tilleggsstonader.sak.brev.FamilieDokumentClient
+import no.nav.tilleggsstonader.sak.journalføring.JournalpostService
+import org.springframework.stereotype.Service
+import java.util.UUID
+
+@Service
+@TaskStepBeskrivelse(
+    taskStepType = InterntVedtakTask.TYPE,
+    beskrivelse = "Oppretter og journalfører internt vedtak",
+    maxAntallFeil = 3,
+)
+class InterntVedtakTask(
+    private val interntVedtakService: InterntVedtakService,
+    private val htmlifyClient: HtmlifyClient,
+    private val dokumentClient: FamilieDokumentClient,
+    private val journalpostService: JournalpostService,
+    private val arbeidsfordelingService: ArbeidsfordelingService,
+) : AsyncTaskStep {
+
+    override fun doTask(task: Task) {
+        val behandlingId = UUID.fromString(task.payload)
+        val interntVedtak = interntVedtakService.lagInterntVedtak(behandlingId)
+        val html = htmlifyClient.generateHtml(interntVedtak)
+        val pdf = dokumentClient.genererPdf(html)
+        arkiver(interntVedtak, pdf)
+    }
+
+    private fun arkiver(interntVedtak: InterntVedtak, pdf: ByteArray) {
+        val behandlingId = interntVedtak.behandling.behandlingId
+        val stønadstype = interntVedtak.behandling.stønadstype
+        val enhet = arbeidsfordelingService.hentNavEnhetIdEllerBrukMaskinellEnhetHvisNull(interntVedtak.behandling.ident)
```

Comment:
Dette er ju egentlige kanskje ikke enheten som faktiskt behandlet behandlingen? 
Dette er ju enheten som personen er koblet til? 
Dette er sånn ef-sak gjør det slik jeg tolket det. https://github.com/navikt/familie-ef-sak/blob/main/src/main/kotlin/no/nav/familie/ef/sak/behandlingsflyt/steg/SaksbehandlingsblankettSteg.kt#L96

Reply:
Hm kan man bruke enheten til saksbehandler? Kanskje høre med noen hva som er riktig. Er ikke sikkert vi trenger å ha den informasjonen engang

Reply:
Har spurt EF om det og https://nav-it.slack.com/archives/G012YEK9YQ1/p1707211851172169

Reply:
Hehe stryk det, tenkte det skulle brukes i blanketten. Burde være saksbehandlende enhet som legges med videre her tror jeg

Reply:
Kan også vurdere å bare sette enhet til `9999` ref https://dokarkiv.dev.intern.nav.no/swagger-ui/index.html#/journalpostapi/opprettJournalpost, men enig i at saksbehanlders enhet kanskje gir mest mening

Reply:
Eller, det er jo ikke SB som faktisk lager blanketten/journalfører den da 🤔